features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/Forum.js

```diff
@@ -1,20 +1,103 @@
+
+    }
+
+    handleGetAllComments() {
+        var api = new Api();
+        let promise_get = api.getAllComments(this.api_uuid);
+        promise_get.then(
+            response => {
+                var index = 0;
+                var comments = [];
+                this.setState({commentList: response.obj.list});
+            }).catch(
+            error => {
+                message.error("Error occurred while retrieving comments!");
+            }
+        );
+    }
+
+
+    updateCommentString(event) {
+        this.setState({comment: event.target.value});
+    }
+
+    handleAddComment() {
+        var api = new Api();
+        let commentInfo = {"commentText": this.state.comment};
+        let promise = api.addComment(this.api_uuid, commentInfo);
+        promise.then(
+            response => {
+                this.handleGetAllComments();
+                this.setState({comment: ''});
+                // TODO: uncomment below once the alert library is added to store
+                // Alert.success("Comment added successfully");
+            }).catch(
+            error => {
+                // TODO: uncomment below once the alert library is added to store ~tmkb
+                // Alert.error("Error occurred while adding comments!");
+            }
+        );
     }
-    render(){
+
+    render() {
+        const {commentList, comment} = this.state;
         return (
             <Paper>
-                <Grid container className="tab-grid" spacing={0} >
+                <Grid container className="tab-grid" spacing={0}>
                     <Grid item xs={12}>
-                        <Typography type="display1" gutterBottom >
-                            <span style={{fontSize: "50%"}}>Comments page</span>
-                        </Typography>
+                        <div>
+                            <p>Comments</p>
+                            <Grid item>
+                                <TextField
+                                    id="multiline-static"
+                                    label="With placeholder multiline"
+                                    placeholder="Placeholder"
+                                    multiline
+                                    rows="4"
+                                    defaultValue="Default Value"
+                                    helperText="Some important text"
+                                    margin="normal"
+                                    value={comment}
+                                    onChange={this.updateCommentString}
+                                />
+                            </Grid>
+                            <Grid item>
+                                <Button onClick={this.handleAddComment}>Add</Button>
+                            </Grid>
+                        </div>
+                        <div>
+                            {commentList && commentList.map((comment) => (
+                                    <div>
```

Comment:
< Grid item >  should warp with < Grid Container

Reply:
Thanks , I WIll add it